### PR TITLE
Add Hub installation 

### PR DESCRIPTION
**Pull request by andy-mishechkin**
According to https://github.com/actions/virtual-environments/issues/36 Hub CLI installation has been added to Windows\Linux image generation scripts.
Builds from branch [v-andmis/36_Add_Hub] have been created successfully:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=61647&view=results
https://github.visualstudio.com/virtual-environments/_build/results?buildId=61692&view=results
https://github.visualstudio.com/virtual-environments/_build/results?buildId=61690&view=results
https://github.visualstudio.com/virtual-environments/_build/results?buildId=61691&view=results
VMs have been created, and Hub has been tested there through next steps:
- Windows and Linux VMs were enabled
- PATH environment variable was checked for correct path to Git and Hub CLI directory
- according to https://hub.github.com/hub.1.html some Hub CLI commands have been executed in VM serial console (hub, hub help, etc)

### CODE REVIEW

**Comment by vmapetr** [comment]
> VMs have been created, and Hub has been tested there.
Could you please provide testing steps?

**Comment by AlenaSviridenko** [comment]
Should we document this installed tool in readme for Windows and Ubuntu?

File: images/linux/scripts/installers/git.sh [hunk 1]
```diff
@@ -35,3 +35,11 @@ echo "Lastly, document the installed versions"
 DocumentInstalledItem "Git ($(git --version 2>&1 | cut -d ' ' -f 3))"
 # git-lfs/2.6.1 (GitHub; linux amd64; go 1.11.1)
 DocumentInstalledItem "Git Large File Storage (LFS) ($(git-lfs --version 2>&1 | cut -d ' ' -f 1 | cut -d '/' -f 2))"
+
+#Install hub
+snap install hub --classic
```
Comment (al-cheb) [inline images/linux/scripts/installers/git.sh]: Please update documentation to provide hub version

File: images/win/scripts/Installers/Install-Git.ps1 [hunk 2]
```diff
@@ -8,6 +8,7 @@ Import-Module -Name ImageHelpers
 # Install the latest version of Git which is bundled with Git LFS.
 # See https://chocolatey.org/packages/git
 choco install git -y --package-parameters="/GitAndUnixToolsOnPath /WindowsTerminal /NoShellIntegration"
+choco install hub
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Git.ps1]: Please update documentation to provide hub version

**Comment by andy-mishechkin** [comment]
> > VMs have been created, and Hub has been tested there.
> > Could you please provide testing steps?

Done

File: images/win/scripts/Installers/Install-Git.ps1 (hunk 2 above)
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Git.ps1]: Add simple validation to check if hub is installed

File: images/win/scripts/Installers/Validate-Git.ps1 [hunk 3]
```diff
@@ -3,18 +3,21 @@
...
 
-if((Get-Command -Name 'git') -and (Get-Command -Name 'bash') -and (Get-Command -Name 'awk') -and (Get-Command -Name 'git-lfs'))
+if((Get-Command -Name 'git') -and (Get-Command -Name 'bash') -and (Get-Command -Name 'awk') -and (Get-Command -Name 'git-lfs') -and (Get-Command -Name 'hub'))
 {
     Write-Host "$(git version) on path"
     Write-Host "$(git-lfs version) on path"
+
+    $strHubVersion = $(hub version)[1]
+    Write-Host "${strHubVersion} on path"
+    $hubVersion = ($strHubVersion -split " ")[2]
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Validate-Git.ps1]: I think we should keep only Write-Host here: `Write-Host "$(hub version | Select-String "hub version") on path"`

Retrieving version should be done below to be consistent with other tools:
```
if( $(hub version) -match  'hub version (?<version>.*)' )
{
   $hubVersion = $Matches.version
}
```

File: images/win/scripts/Installers/Validate-Git.ps1 [hunk 4]
```diff
@@ -26,24 +29,35 @@ if( $(git-lfs version) -match  'git-lfs\/(?<version>.*) \(Git.*' )
...
+    [pscustomobject]@{
+        SoftwareName = "Hub CLI";
+        Description = @"
+_Version:_ $hubVersion<br/>
+_Environment:_
+* PATH: contains location of hub.exe
+* GIT_LFS_PATH: location of hub.exe
+"@
+    }
+)
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-Git.ps1]: This section for my mind contains very difficult formatting style. Shorter variant:

```
# Adding description of the software to Markdown
$GitSoftwareDescription = @"
_Version:_ $gitVersion<br/>
_Environment:_
* PATH: contains location of git.exe
"@

Add-SoftwareDetailsToMarkdown -SoftwareName "Git" -DescriptionMarkdown $GitSoftwareDescription

$GitLFSDescription = @"
_Version:_ $gitLfsVersion<br/>
_Environment:_
* PATH: contains location of git-lfs.exe
* GIT_LFS_PATH: location of git-lfs.exe
"@

Add-SoftwareDetailsToMarkdown -SoftwareName "Git Large File Storage (LFS)" -DescriptionMarkdown $GitLFSDescription

$HubCliDescription = @"
_Version:_ $hubVersion<br/>
_Environment:_
* PATH: contains location of hub.exe
* GIT_LFS_PATH: location of hub.exe
"@

Add-SoftwareDetailsToMarkdown -SoftwareName "Hub CLI" -DescriptionMarkdown $HubCliDescription

```

File: images/linux/scripts/installers/git.sh (hunk 1 above)
Comment (andy-mishechkin) [reply images/linux/scripts/installers/git.sh]: Done: https://github.com/andy-mishechkin/virtual-environments/commit/9eb8d744a7f31c7d839aa8ea6a5fc21ca0147e5c
Builds:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62031&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62030&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62028&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62029&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43

File: images/win/scripts/Installers/Install-Git.ps1 (hunk 2 above)
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Install-Git.ps1]: Done: https://github.com/andy-mishechkin/virtual-environments/commit/9eb8d744a7f31c7d839aa8ea6a5fc21ca0147e5c
Builds:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62031&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62030&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62028&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62029&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43

Comment (andy-mishechkin) [reply images/win/scripts/Installers/Install-Git.ps1]: Done: https://github.com/andy-mishechkin/virtual-environments/commit/9eb8d744a7f31c7d839aa8ea6a5fc21ca0147e5c
Builds:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62031&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62030&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62028&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62029&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43

Comment (andy-mishechkin) [reply images/win/scripts/Installers/Install-Git.ps1]: Done: https://github.com/andy-mishechkin/virtual-environments/commit/9eb8d744a7f31c7d839aa8ea6a5fc21ca0147e5c
Builds:
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62031&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62030&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62028&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43
https://github.visualstudio.com/virtual-environments/_build/results?buildId=62029&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=486c0909-f67d-5748-80b3-bb1d531e7b43

File: images/win/scripts/Installers/Validate-Git.ps1 (hunk 4 above)
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: It's  copy-past. I don't think that's right

Comment (al-cheb) [reply images/win/scripts/Installers/Validate-Git.ps1]: Not in this case. It's longer and very poor formatted style.

File: images/win/scripts/Installers/Validate-Git.ps1 (hunk 3 above)
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: Done:
https://github.com/andy-mishechkin/virtual-environments/commit/684f2a4e0134e768185991a7a9ef0fe9f6c91bbd

File: images/win/scripts/Installers/Validate-Git.ps1 [hunk 5]
```diff
@@ -25,25 +25,41 @@ if( $(git-lfs version) -match  'git-lfs\/(?<version>.*) \(Git.*' )
...
-
-$Description = @"
+    },
+    [pscustomobject]@{
+        SoftwareName = "Git Large File Storage (LFS)";
+        Description = @"
 _Version:_ $gitLfsVersion<br/>
 _Environment:_
 * PATH: contains location of git-lfs.exe
 * GIT_LFS_PATH: location of git-lfs.exe
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-Git.ps1]: Where do we set the GIT_LFS_PATH variable?

File: images/win/scripts/Installers/Validate-Git.ps1 [hunk 6]
```diff
@@ -25,25 +25,41 @@ if( $(git-lfs version) -match  'git-lfs\/(?<version>.*) \(Git.*' )
...
 "@
+    },
+    [pscustomobject]@{
+        SoftwareName = "Hub CLI";
+        Description = @"
+_Version:_ $hubVersion<br/>
+_Environment:_
+* PATH: contains location of hub.exe
+* GIT_LFS_PATH: location of hub.exe
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-Git.ps1]: Why do we put this to HUB description?

Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: removed

File: images/win/scripts/Installers/Validate-Git.ps1 (hunk 5 above)
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: I didn't find setting this variable somewhere in install scripts and in VMs, but this variable may used in some builds. I suggest to leave it here

Comment (al-cheb) [reply images/win/scripts/Installers/Validate-Git.ps1]: Please provide an example output from Windows Server agents where the `GIT_LFS_PATH`  variable is defined.

File: images/win/scripts/Installers/Validate-Git.ps1 (hunk 4 above)
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: https://github.com/andy-mishechkin/virtual-environments/commit/9f7ab8176cb072b05ff08150f8fd8221a3cdfb52

File: images/win/scripts/Installers/Validate-Git.ps1 [hunk 7]
```diff
@@ -3,18 +3,18 @@
 ##  Desc:  Validate Git for Windows
 ################################################################################
 
-if((Get-Command -Name 'git') -and (Get-Command -Name 'bash') -and (Get-Command -Name 'awk') -and (Get-Command -Name 'git-lfs'))
```
[the remark this answers is not in the document]
Comment (andy-mishechkin) [reply images/win/scripts/Installers/Validate-Git.ps1]: Total refactoring has been done:
https://github.com/andy-mishechkin/virtual-environments/commit/fa69a832803433b6257b3807d5f8167414c44987

**Pull request merged**